src/OpenLoco/Windows/TerraForm.cpp

```diff
@@ -1345,6 +1348,11 @@ namespace OpenLoco::Ui::Windows::Terraform
                     self->invalidate();
                     break;
                 }
+
+                case widx::paint_mode:
```

Comment:
Shouldn't the button be left in a pressed state? Then you could use that to interpret if its in use without resorting to adding isPaintMode.

Reply:
It is left in a pressed state. `onMouseDown` is the event for when the mouse button transitions from up to down whilst over the button. The `isPaintMode` is the actual thing which stores the up/down state of the button, as buttons themselves don't seem to do this, unless I've missed something.